registry/session.go

```diff
@@ -222,10 +222,10 @@ func (r *Session) GetRemoteTags(registries []string, repository string, token []
 		logrus.Debugf("Got status code %d from %s", res.StatusCode, endpoint)
 		defer res.Body.Close()
 
-		if res.StatusCode != 200 && res.StatusCode != 404 {
-			continue
-		} else if res.StatusCode == 404 {
+		if res.StatusCode == 404 {
 			return nil, fmt.Errorf("Repository not found")
+		} else if res.StatusCode != 200 {
```

Comment:
I think these should be separate, `else`'s are difficult to read, and it's technically redundant since you're returning in the first branch.